components-api/src/main/scala/pl/touk/nussknacker/engine/api/Context.scala

```diff
@@ -1,35 +1,43 @@
 package pl.touk.nussknacker.engine.api
 
-import java.util.UUID
-import scala.util.Random
+import scala.jdk.CollectionConverters._
 
 object Context {
 
-  // prefix is to distinguish between externally provided and internal (initially created) id
-  private val initialContextIdPrefix = "initial-"
-
-  /**
-   * For performance reasons, is used unsecure random - see UUIDBenchmark for details. In this case random correlation id
-   * is used only for internal purpose so is not important in security context.
-   */
-  private val random = new Random()
-
-  /**
-   * Deprecated: should be used ContextIdGenerator e.g. via EngineRuntimeContext.contextIdGenerator
-   * Should be used for newly created context - when there is no suitable external correlation / tracing id
-   */
-  def withInitialId: Context = {
-    Context(initialContextIdPrefix + new UUID(random.nextLong(), random.nextLong()).toString)
-  }
-
-  def apply(id: String): Context = Context(id, Map.empty, None)
+  def apply(id: ContextId): Context = Context(id, Map.empty, None)
 
-  def apply(id: String, variables: Map[String, Any]): Context =
+  def apply(id: ContextId, variables: Map[String, Any]): Context =
     Context(id, variables, None)
 
+  def dummy: Context = Context(ContextId.dummy)
```

Comment:
Let's write a comment that it is for tests purpose. After we change `EventGeneratorSourceFactory` it will be true. Won't be?